[IMP] iot: esc-pos ESC * support documentation (#5367)
odoo/documentation: content/applications/productivity/iot/config/troubleshooting.rst

   Epson and Star receipt printers and Zebra label printers do not need a driver to work. Make sure
   that no driver is selected for those printers.

> Epson special case

Reviewer: There isn't an error message populated. There are only garbled printouts. This is labeled in the previous heading.  I've added in the "configuration" special case, as this is what I would search.

~~~~~~~~~~~~~~~~~~

When printing a receipt with an Epson printer, the command that is used is called `GS v 0`. However,
there are some Epson printers models that do not support this command, they include the following

> models:

Reviewer: I've added this in. Thanks for the suggestion.

- TM-U220
- TM-U230
- TM-P60
- TMP-P60II

If the :abbr:`POS (Point of Sale)` system is using one of these printer models, a particular
process must be undertaken in order to use another command; `ESC *`.

Process to force ESC * command
******************************

The first step is to check whether the printer is incompatible with `GS v 0` command.

.. seealso::
   See this `Epson's GS v 0 documentation
   <https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=94>`_ for `GS v 0`
   compatible printers.

Should the printer not be compatible with the `GS v 0` command ensure that it is compatible with the
`ESC *` command.

.. seealso::
   See this `Epson ESC * documentation
   <https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=88>`_ for `ESC *`
   compatible printers.

If the printer is not compatible with the `ESC *` command then the following process is
not possible. Should the printer be compatible to use the `ESC *` command to print, follow this

> process to configure the printer with the :abbr:`IoT (Internet of Things)` box.

Reviewer: I've made an H7 heading to include this section separately. Each section is split out and has a "Table of Contents" on the right-hand column made from the headings. Great suggestions.  Thanks! :1st_place_medal:

To manipulate the :abbr:`IoT (Internet of Things)` box to use the `ESC *` command to print, go to
the :abbr:`IoT (Internet of Things)` box homepage. Access the :abbr:`IoT (Internet of Things)` box

> homepage by navigating to :menuselection:`IoT app --> IoT Box`. Then click on the ::guilabel:`IP

Reviewer: This should be labeled IoT Boxes. Nice catch! I've resolved this issue.

address` and this will redirect the page to the :abbr:`IoT (Internet of Things)` box homepage.

Now click on the :guilabel:`Printers server` button. This will redirect the browser to the *CUPS

> page*. Next, go to :menuselection:`Administration --> Printers --> Add Printer`. Now choose the

Reviewer 2: CUPS allows computers to act as a printing server. 
Yes a definition would be nice but it's not necessary to ship right now.
Resolving.

printer that should be modified and click :guilabel:`Continue`.

.. tip::
   If the name of the printer is still uncertain, take the following steps:

   #. Take note of the listed printers on the *CUPS* page.
   #. Turn the printer off and refresh the page.
   #. Now compare the difference with the first list to see which printer disappeared.
   #. Turn the printer back on and refresh the page again.
   #. Double check the list again to see if the printer re-appears.
   #. The printer that disappeared and reappears again on the listed printers is the name of the
      printer in question.

   .. note::
      This can be :guilabel:`Unknown` under :guilabel:`Local printers`.

> 

Reviewer 3: What do you think about adding a new h5 heading here? We could name it something like 

```suggestion
Configure CUPS
*************
```

After setting the name of the printer with the appropriate naming convention, click
:guilabel:`Continue`. Next, set the :guilabel:`Make` value to :guilabel:`Raw` and for the
:guilabel:`Model` value, set this to :guilabel:`Raw Queue (en)`.

After completing these steps, click :guilabel:`Add Printer`. If everything was done correctly, the
page should redirect to the *Banners* page.

At this point the printer should have been created, now the :abbr:`IoT (Internet of Things)` box
just needs to detect it and then sync to Odoo's server (this could take a few minutes).

Once the printer is visible on the Odoo database, don't forget to choose it in the :abbr:`PoS (Point
of Sale)`configuration as the :abbr:`IoT (Internet of Things)` printer. Navigate to
:menuselection:`Pos App --> Settings --> Connected Devices --> IoT Box --> Receipt Printer -->
Save`.

.. note::
   If the printer was set up incorrectly (it is still printing random text or the printed receipt is
   too big or small) it cannot be modified via the printer name with *CUPS*. Instead the above
   process can be repeated to set up another printer from scratch to create one with modified
   parameters.

> .. spoiler::

Reviewer 3: What do you think of adding an h5 header titled "Example: configure TM-U220B printer"? Also the info you shared in this spoiler block seems too valuable to hide in a hidden block like this. Of the contents inside this block, is there a way to take 1 or 2 screenshots out of here to place in an example admonition block in the earlier sections? (:

Reviewer: I added an H7 heading to this section. I will leave it as a spoiler because it is too specific to general troubleshooting and takes up too much real estate. This is a creative way to use the spoiler admonition block and hide unnecessary info. Because this is a practical example, the images are justified.

   of the :guilabel:`Discovered Network Printers`. Instead it is likely part of the
   :guilabel:`Unknown` selection. By unplugging the printer's :abbr:`USB (Universal Serial Bus)`
   cable from the :abbr:`IoT (Internet of Things)` box and refreshing the page, the
   :guilabel:`Unknown` printer disappears. By plugging it back in, then it reappears, so it can be
   said that this is the printer in question.

   For the naming convention, since it will need to print using the `ESC *` command, it is
   imperative to add `__IMC`. Reference the printer model on `Epson's ESC * site
   <https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=88>`_ to find out more
   about the *density* parameters.

   .. image:: troubleshooting/epson-TM-U220-specification.png
      :align: center
      :alt: Epson TM-U220 specifications on manufacturers website.

   For this particular model, TM-U220, `m` should be equal to 0 or 1. While referencing the
   :guilabel:`Description` below the pink box in the above picture, the `m` values could be 0, 1, 32
   or 33. So in this printers case, the `m` value can NOT be 32 or 33 (otherwise there will be
   random characters printed).

>    If we check the table for the values: 32 and 33, they both occur if the :guilabel:`Number of bits

Reviewer 3: We don't use pronouns like "we" in docs :astonished: Can you rewrite all instances of this, please? (:

Reviewer: Noted, this is the only instance. Thank you.

   Epson and Star receipt printers and Zebra label printers do not need a driver to work. Make sure
   that no driver is selected for those printers.

Epson special case
~~~~~~~~~~~~~~~~~~

When printing a receipt with an Epson printer, the command that is used is called `GS v 0`. However,
there are some Epson printers models that do not support this command, they include the following
models:

- TM-U220
- TM-U230
- TM-P60
- TMP-P60II

If the :abbr:`POS (Point of Sale)` system is using one of these printer models, a particular
process must be undertaken in order to use another command; `ESC *`.

Process to force ESC * command
******************************

The first step is to check whether the printer is incompatible with `GS v 0` command.

.. seealso::
   See this `documentation on Epison's site

>    <https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=94>`_ for compatible

Author: ```suggestion
   <https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=94>`_ for `GS v 0` compatible
```

Author: I'm afraid that if client see that it looks the same, they will assume it is the same link

Reviewer: Noted,  I've made the change.

   printers.

Should the printer not be compatible with the `GS v 0` command ensure that it is
compatible with the `ESC *` command.

.. seealso::
   See the `Epson documentation

>    <https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=88>`_ for compatible

Author: ```suggestion
   <https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=88>`_ for `ESC *` compatible
```

Author: the "see also" above mention "documentation on Epison's site" while here we write "Epson documentation", maybe we should be more consistent ?
Could be a good occasion to merge both ideas:
` See the "Epson ESC* documentation" <https://..."` for this one and ` See the "Epson GS V 0 documentation" <https://..."` for the first ?

Reviewer: @lse-odoo 

We get the following build error when the reference text is the same: 
`
/home/tiku/github/odoo/documentation/content/applications/productivity/iot/config/troubleshooting.rst:3: WARNING: Duplicate explicit target name: "documentation on epison's site".`

Reviewer: I spoke with our Web Designer (@samueljlieber), and we came to the conclusion that making a small difference in the text would be helpful so we don't get any build errors in the long-run.

   printers.

If the printer is not compatible with the `ESC *` command then the following process is
not possible. Should the printer be compatible to use the `ESC *` command to print, follow this
process to configure the printer with the :abbr:`IoT (Internet of Things)` box.

To manipulate the :abbr:`IoT (Internet of Things)` box to use the `ESC *` command to print, go to
the :abbr:`IoT (Internet of Things)` box homepage. Access the :abbr:`IoT (Internet of Things)` box
homepage by entering the :abbr:`IP (Internet Protocol)` address of the :abbr:`IoT (Internet of
Things)` box into a browser window from a computer on the same network as the :abbr:`IoT (Internet

> of Things)` box (preferably by ethernet connection).

Author: @tiku-odoo is there not another part of the documentation which already exist to explain how access the homepage that we can refer instead ? I'm afraid that it might confuse people "Ip address" when they can "simply" access it through Odoo IoT app

Reviewer: @lse-odoo  

Nice catch! I have made the change to simplify this process to: going to the IOT app --> IoT box --> Click on IP. 

Thanks for suggesting this change.

.. note::
   The :abbr:`IP (Internet Protocol)` address can be accessed by the router admin console that the
   :abbr:`IoT (Internet of Things)` box is connected to, or by connecting a working receipt printer
   to the :abbr:`IoT (Internet of Things)` box. A receipt will print out with the :abbr:`IoT

>    (Internet of Things)` box's :abbr:`IP (Internet Protocol)` address on it.

Author: @tiku-odoo  
+ it might be even more confusing as, if I am not mistaken, if the printer is in the network it will print it's IP address (and not the one of the IoT box). Generally they will be connected in USB, but it might not always be the case.

++ in the case of the TMU220 it is a thermal printer so it will not print any ticket (I think)

Reviewer: I've removed this section from the doc. This change won't be necessary. Thank you! 

Tim

Then click on the :guilabel:`Printers server` button. This will redirect the browser to the *CUPS
page*. Next, go to :menuselection:`Administration --> Printers --> Add Printer`. Now choose the

> printer that should be modified. It could be listed as :guilabel:`Unknown`. Click

Author: "It could be listed as :guilabel:`Unknown`" is redundant with the note below

Reviewer: Noted, I've removed the upper instance from the doc. Thanks for spotting this.

:guilabel:`Continue`.

.. tip::
   If the name of the printer is still uncertain, take the following steps:

   #. Take note of the listed printers on the *CUPS* page.
   #. Turn the printer off and refresh the page.
   #. Now compare the difference with the first list to see which printer disappeared.
   #. Turn the printer back on and refresh the page again.
   #. Double check the list again to see if the printer re-appears.
   #. The printer that disappeared and reappears again on the listed printers in the name of the
      printer in question.

   .. note::

>       This can be :guilabel:`Unknown` under local printers.

Author: "local printers" might need a :guilabel: or something as it refers to the name of an option (see screenshot in  the example with Add Printer instruction

Reviewer: Good call, thank you.

`CUPS` will prompt the admin for three pieces of information: the :guilabel:`Name`,

> :guilabel:`Description` and the :guilabel:`Location`. The last to pieces of information do not need

Author: "The last to pieces of information" redundant "to"

Author: or probably mean "two" ?

Reviewer: Exactly, thank you! Made the change!

to be specific, however, the :guilabel:`Name` should follow a particular convention to work with the
`ESC *` command.

The :guilabel:`Name` should match this convention:
`<printer_name>__IMC_<param_1>_<param_2>_..._<param_n>__`

A breakdown of the naming convention:

- `printer_name`: Is the printer name. It can be any character as long as it does not contain `_`,
  `/`, `#`, or ` ` (space character).
- `IMC`: Stands for Image Mode Column (the simplified name for `ESC *`).
- `param_1`: specific parameter:

  - `SCALE<X>`: Scale of the picture (with the same aspect ratio). `X` should be an integer
    describing the scale percentage that should be used.

    .. example::
       `100` is the original size, `50` is half the size, `200` is twice the size.

  - `LDV`: Low Density Vertical (will be set to High Density Vertical if not specified).
  - `LDH`: Low Density Horizontal (will be set to High Density Horizontal if not specified).

    .. note::
       "Density" parameters might need to be configured in a particular way depending on the printer
       model.

.. seealso::
   Visit `Epson's documentation

>    <https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=88>`_ and click on the

Author: same as above, consider changing it to be more explicit: "Epson's ESC * documentation"

Reviewer: Noted, I've made a change.

   printer model printer in the table above to see if the printer should set these parameters.

.. example::

>    The follow are examples of proper and improper name formatting:

Author: can we write "The follow ..." ? I thought it was "The following..."

Reviewer: Yes this is correct. I've made the change. Thank you! @lse-odoo

After setting the name of the printer with the appropriate naming convention, click
:guilabel:`Continue`. Next, set the :guilabel:`Make` value to :guilabel:`Raw` and for the
:guilabel:`Model` value, set this to :guilabel:`Raw Queue (en)`.

After completing these steps, click :guilabel:`Add Printer`. If everything was done correctly, the
page should redirect to the "Banners" page.

At this point the printer should have been created, now the :abbr:`IoT (Internet of Things)` box
just needs to detect it and then sync to Odoo's server (could take a few minutes).

Once the printer is visible on the Odoo database, don't forget to choose it in the PoS configuration
as the :abbr:`IoT (Internet of Things)` printer. :menuselection:`Pos App --> Settings -->
Connected Devices --> IoT Box --> Receipt Printer --> Save`.

.. note::
   If the printer was set up incorrectly (still printing random text or the printed receipt is too
   big or small) it cannot be modified via the printer name with CUPS. Instead the above process can
   be repeated to set up another printer from scratch to create one with modified parameters.

.. spoiler::

>    A example tutorial for the Epson TM-U220B printer model.

Author: A or An ?

   said that this is the printer in question.

   For the naming convention, since it will need to print using the `ESC *` command, it is
   imperative to add `__IMC`. With regard to the *density* parameters, toggle to the printer model
   on `Epson's website
   <https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=88>`_.

   .. image:: troubleshooting/epson-TM-U220-specification.png
      :align: center
      :alt: Epson TM-U220 specifications on manufacturers website.

   For this particular model, TM-U220, `m` should be equal to 0 or 1. While referencing the
   :guilabel:`Description` below the pink box in the above picture, the `m` values could be 0, 1, 32
   or 33. So in this printers case, the `m` value can NOT be 32 or 33 (otherwise there will be
   random characters printed).

   .. example::
      If we check the table for the values: 32 and 33, they both occur if the :guilabel:`Number of
      bits for vertical data` is set to 24. This means that is a *High Vertical Density*. In the
      case of configuring the Epson TM-U220, the *Low Vertical Density* will need to be forced, as

>       this printer model does not support "High Vertical Density" for this command.

Author: A bit strange IMHO as the "example" depends on the part before which is already an example itself. I don't really see the benefit of splitting the explanations partially in an example. I do agree that it could probably be improve (maybe in a "detailed" section ?), but I'm not convinced with the current way personally

Reviewer: Sounds better to me without the admonition block as well. Thanks for your honest opinion.

   To add a *Low Vertical Density* add the `LDV` parameter to the naming convention.

   .. image:: troubleshooting/add-printer-filled.png
      :align: center
      :alt: Add a *Low Vertical Density* add the `LDV` parameter to the naming convention.

   Click :guilabel:`Continue` to proceed. Next, set the :guilabel:`Make` value to :guilabel:`Raw`
   and for the :guilabel:`Model` value, set this to :guilabel:`Raw Queue (en)`.

   .. image:: troubleshooting/add-printer-add.png
      :align: center
      :alt: Epson TM-U220 specifications on manufacturers website.

>    However when trying to print with `EpsonTMU220B__IMC_LDV__` as the naming convention  it will

Author: the formulation is a bit odd or it is me ? maybe with "as the naming convention is respected/followed" ?

Reviewer: Noted, I've reworked this section a bit.

Most Epson printers support printing receipts in Odoo :abbr:`POS (Point of Sale)` using the `GS v 0`
command. However, the following Epson printer models do not support this command:

- TM-U220
- TM-U230
- TM-P60
- TMP-P60II

Bypass this issue by configuring the printer to use the `ESC *` command instead.

Process to force ESC * command
******************************

Epson printer compatibility
^^^^^^^^^^^^^^^^^^^^^^^^^^^

The first step is to check whether the printer is incompatible with `GS v 0` command.

.. seealso::
   See this `Epson's GS v 0 documentation
   <https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=94>`_ for `GS v 0`
   compatible printers.

   See this `Epson ESC * documentation
   <https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=88>`_ for `ESC *`
   compatible printers.

If the printer is not compatible with the `ESC *` command then the following process is
not possible. Should the printer be compatible to use the `ESC *` command to print, follow this
process to configure the printer with the :abbr:`IoT (Internet of Things)` box.

IoT box configuration for ESC *
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> To manipulate the :abbr:`IoT (Internet of Things)` box to use the `ESC *` command to print, go to

Reviewer 4: ```suggestion
To configure the :abbr:`IoT (Internet of Things)` box to use the `ESC *` command to print, go to
```
This seems like the only use of "manipulate" instead of "configure", so it seems a bit out of place unless there is reason for it.

   - `EPSONTMm30II__IMC_SCALE__` -> The parameter `SCALE` is missing the scale value.

Finish adding a printer
"""""""""""""""""""""""

After setting the name of the printer with the appropriate naming convention, click
:guilabel:`Continue`. Next, set the :guilabel:`Make` value to :guilabel:`Raw` and for the
:guilabel:`Model` value, set this to :guilabel:`Raw Queue (en)`.

After completing these steps, click :guilabel:`Add Printer`. If everything was done correctly, the
page should redirect to the *Banners* page.

At this point the printer should have been created, now the :abbr:`IoT (Internet of Things)` box
just needs to detect it and then sync to Odoo's server (this could take a few minutes).

Adding the printer to Odoo PoS
""""""""""""""""""""""""""""""

Once the printer is visible on the Odoo database, don't forget to choose it in the :abbr:`PoS (Point
of Sale)`configuration as the :abbr:`IoT (Internet of Things)` printer. Navigate to
:menuselection:`Pos App --> Settings --> Connected Devices --> IoT Box --> Receipt Printer -->
Save`.

.. note::
   If the printer was set up incorrectly (it is still printing random text or the printed receipt is

>    too big or small) it cannot be modified via the printer name with *CUPS*. Instead the above

Reviewer 4: ```suggestion
   too big or small) it cannot be modified via the printer name with *CUPS*. Instead, the above
```

   process can be repeated to set up another printer from scratch to create one with modified
   parameters.

Example setup of the Epson TM-U220B printer using ESC *
"""""""""""""""""""""""""""""""""""""""""""""""""""""""

.. spoiler::
   Click arrow to reveal.

   An example tutorial for the Epson TM-U220B printer model.

>    The following is an example of the troubleshooting process for a TM-U220B printer model to use

Reviewer 4: ```suggestion
   The following is an example of the troubleshooting process for a TM-U220B printer model using
```

>    the `ESC *` command. The receipt picture below is an example of a receipt that is printing

Reviewer 4: ```suggestion
   the `ESC *` command. The receipt pictured below is an example of a receipt that is printing
```
Either "receipt pictured below" or just "picture below" I think would work better here.

Reviewer: Thanks! Good catch

   correctly due to proper formatting (in theory):

   .. image:: troubleshooting/receipt-example.png
      :align: center
      :alt: Properly formatted receipt picture from a demo database.

>    Trying to print this receipt right-away prior to the proper formatting  will not work as the

Reviewer 4: ```suggestion
   Trying to print this receipt right-away prior to the proper formatting will not work as the
```

   .. image:: troubleshooting/receipt-print-random-letters.png
      :align: center
      :alt: Printer paper with seemingly random characters.

   To properly configure formatting for the Epson TM-U220B printer model take the following steps.

   After consulting Epson's website for compatibility for both of the commands: `GS v 0` and
   `ESC *`, it can be seen that indeed the TM-U220B is not compatible with `GS v 0`, but is
   compatible with `ESC *`.

   .. image:: troubleshooting/epson-compatibility-compare.png
      :align: center
      :alt: Epson compatibility evaluation from Epson website.

   When adding the printer, *CUPS* will ask which printer should be added:

   .. image:: troubleshooting/add-printer.png
      :align: center
      :alt: Administration menu, add printer selection.

   In this case, the printer is connected via :abbr:`USB (Universal Serial Bus)` so it won' be part
   of the :guilabel:`Discovered Network Printers`. Instead it is likely part of the
   :guilabel:`Unknown` selection. By unplugging the printer's :abbr:`USB (Universal Serial Bus)`
   cable from the :abbr:`IoT (Internet of Things)` box and refreshing the page, the

>    :guilabel:`Unknown` printer disappears. By plugging it back in, then it reappears, so it can be

Reviewer 4: ```suggestion
   :guilabel:`Unknown` printer disappears. By plugging it back in, the printer reappears, so it can be
```

   Epson and Star receipt printers and Zebra label printers do not need a driver to work. Make sure
   that no driver is selected for those printers.

Epson configuration special case
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Most Epson printers support printing receipts in Odoo :abbr:`POS (Point of Sale)` using the `GS v 0`
command. However, the following Epson printer models do not support this command:

- TM-U220
- TM-U230
- TM-P60
- TMP-P60II

Bypass this issue by configuring the printer to use the `ESC *` command instead.

Process to force ESC * command
******************************

Epson printer compatibility
^^^^^^^^^^^^^^^^^^^^^^^^^^^

The first step is to check whether the printer is incompatible with `GS v 0` command.

> .. seealso::
>    See this `Epson's GS v 0 documentation
>    <https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=94>`_ for `GS v 0`
>    compatible printers.
> 
>    See this `Epson ESC * documentation
>    <https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=88>`_ for `ESC *`
>    compatible printers.

Reviewer 2: ```suggestion
.. seealso::
   - `Epson GS v 0 documentation <https://reference.epson-biz.com/modules/ref_escpos/index.php?
   content_id=94>`_ for `GS v 0` compatible printers.
   - `Epson ESC * documentation <https://reference.epson-biz.com/modules/ref_escpos/index.php
   ?content_id=88>`_ for `ESC *` compatible printers.
```

If the printer is not compatible with the `ESC *` command then the following process is not
possible. Should the printer be compatible to use the `ESC *` command to print, follow this process
to configure the printer with the :abbr:`IoT (Internet of Things)` box.

IoT box configuration for ESC *
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> To configure the :abbr:`IoT (Internet of Things)` box to use the `ESC *` command to print, go to the
> :abbr:`IoT (Internet of Things)` box homepage. Access the :abbr:`IoT (Internet of Things)` box
> homepage by navigating to :menuselection:`IoT app --> IoT Boxes`. Then click on the :guilabel:`IP
> address` and this will redirect the page to the :abbr:`IoT (Internet of Things)` box homepage.

Reviewer 2: ```suggestion
To configure the :abbr:`IoT (Internet of Things)` box to use the `ESC *` command to print, go to the
:abbr:`IoT (Internet of Things)` box homepage by navigating to :menuselection:`IoT app --> IoT
Boxes`. Then click on the :guilabel:`IP address` and this will direct to the :abbr:`IoT (Internet of
Things)` box homepage.
```

Choosing the printer
""""""""""""""""""""

> Now click on the :guilabel:`Printers server` button. This will redirect the browser to the *CUPS
> page*. Next, go to :menuselection:`Administration --> Printers --> Add Printer`. Now choose the
> printer that should be modified and click :guilabel:`Continue`.

Reviewer 2: ```suggestion
Now click on the :guilabel:`Printers server` button. This will redirect the browser to the *CUPS*
page. Next, go to :menuselection:`Administration --> Printers --> Add Printer`, choose the printer
that should be modified, and then click :guilabel:`Continue`.
```

.. tip::
   If the name of the printer is still uncertain, take the following steps:

   #. Take note of the listed printers on the *CUPS* page.
   #. Turn the printer off and refresh the page.
   #. Now compare the difference with the first list to see which printer disappeared.
   #. Turn the printer back on and refresh the page again.

>    #. Double check the list again to see if the printer re-appears.

Reviewer 2: ```suggestion
   #. Double-check the list again to see if the printer re-appears.
```

   - `EPSONTMm30II__IMC_SCALE__` -> The parameter `SCALE` is missing the scale value.

Finish adding a printer
"""""""""""""""""""""""

After setting the name of the printer with the appropriate naming convention, click
:guilabel:`Continue`. Next, set the :guilabel:`Make` value to :guilabel:`Raw` and for the
:guilabel:`Model` value, set this to :guilabel:`Raw Queue (en)`.

After completing these steps, click :guilabel:`Add Printer`. If everything was done correctly, the
page should redirect to the *Banners* page.

At this point the printer should have been created, now the :abbr:`IoT (Internet of Things)` box
just needs to detect it and then sync to Odoo's server (this could take a few minutes).

Adding the printer to Odoo PoS
""""""""""""""""""""""""""""""

Once the printer is visible on the Odoo database, don't forget to choose it in the :abbr:`PoS (Point
of Sale)`configuration as the :abbr:`IoT (Internet of Things)` printer. Navigate to
:menuselection:`Pos App --> Settings --> Connected Devices --> IoT Box --> Receipt Printer -->
Save`.

.. note::
   If the printer was set up incorrectly (it is still printing random text or the printed receipt is

>    too big or small) it cannot be modified via the printer name with *CUPS*. Instead, the above
>    process can be repeated to set up another printer from scratch to create one with modified

Reviewer 2: ```suggestion
   too big or small), then it cannot be modified via the printer name with *CUPS*. Instead, the
   above process can be repeated to set up another printer from scratch to create one with modified
```

   parameters.

Example setup of the Epson TM-U220B printer using ESC *

> """""""""""""""""""""""""""""""""""""""""""""""""""""""

Reviewer 2: The content in this section is relevant however it's a bit ugly and breaks on mobile.

Reviewer 2: Would like to revisit the use of quotation marks as "h7" formatting—

Typically headings only go down to h6 which are normally associated with subtext (think footer links), so we'd need to create a specific style for this if we were to use it.

As it is, out of the box, the use of quotation mark symbols is reserved to indicate [paragraph text in Sphinx](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections).

...so it might actually just be better to use **bold text** because, as it is now, the `"` formatting doesn't render anything  visually unique enough to say to the reader "hey I'm a heading!"

Reviewer 2: Also we cannot merge the doc with invalid characters like this as the error will propagate to all other versions of the doc across multiple versions (in effect, replicating an error).

Will replace with **bold text** for now.
